### PR TITLE
Add onChange props to React forms

### PR DESCRIPTION
**Pull request by Lily418**
This is for issue #1271 and an alternative to #1446

Opening a new PR as I've removed any changes to core.

# Possible Examples
I intend to update the example for something which shows a use of this feature. 
## Updating Error
Changing the validation logic as discussed here
https://spectrum.chat/jsonforms/general/pristine-and-dirty-checking~2ece93ab-7783-41cb-8ba1-804414eb1da4

or localising the error as discussed
https://spectrum.chat/jsonforms/general/development-sharing-state-between-reducers~d8dfcc0b-7876-4c32-a230-baff42501cab

Both these cases require an additional action for updating errors

### CODE REVIEW

**Comment by Lily418** [comment]
Hi @edgarmueller I've followed your suggestion for onChange handler. 

However at the moment I have an issue, the onChange handler in the onChange example dispatches the UpdateErrors actions asynchronous. This means as you are typing the error flashes up each key press before it is removed. 

I'm unsure of the solution to this, I wonder if rather than dispatching an action the onChange handler should return the data and errors to render. This way the core state in unalterered by gives the user a hook to change what gets renderered.

**Comment by edgarmueller** [comment]
Right, there seems to be an issue when updating the errors, but anyways: I'd suggest to split these changes into two PRs:
1. one which is only adding the `onChange` property to the `JsonForms` component as this seems to work fine already :+1:  (you can update this PR)
2. a second one would provide the action for updating the errors state. Also, I think we should not return an action from `onChange` but instead updating the errors on the client side, i.e. it's up to the consumer to trigger the action within his/her implementation of `onChange` (our current example structure might not properly support this use-case). We can deal with any issues we might face then.

**Comment by Lily418** [comment]
@edgarmueller sounds good, have updated this PR

File: packages/react/src/JsonForms.tsx [hunk 1]
```diff
@@ -48,6 +49,10 @@ interface JsonFormsRendererState {
     resolvedSchema: JsonSchema;
 }
 
+interface JsonFormsReactProps {
+  onChange?(state: {data: any,   errors?: ErrorObject[]}): void;
```
Comment (edgarmueller) [inline packages/react/src/JsonForms.tsx]: We can refine this type via `Pick<JsonFormsCore, 'data' | 'errors'>`

**Comment by Lily418** [comment]
@edgarmueller Thanks for the feedback, Pick type is very nice. have updated.

**Pull request merged**